### ISSUE TITLE
Change Playwright folder structure 

### ISSUE THREAD

**Issue by ProchaLu** [upleveled/next-js-example-spring-2022#7]
To make Playwright test folders more visible, I would change the structure to a similar structure like Cypress because for students, the default Playwright folder structure is not the easiest to read when the project has more files and folders. Playwright also creates two folders for the `report` and `test-results` that would be in the root, but since I would create a folder for all Playwright-related folders and files, I would add them to the folder.

<img width="316" alt="Screenshot 2022-07-27 at 12 18 43" src="https://user-images.githubusercontent.com/80746311/181223838-e8622149-9d18-4633-b1c5-32ff1c7fc814.png">

For the naming, I would name the main folder `playwright` because the default name is 'tests', which is not descriptive enough.

The new structure would look like this in the project: 

<img width="313" alt="Screenshot 2022-07-27 at 14 38 55" src="https://user-images.githubusercontent.com/80746311/181249082-d9e933e7-db87-463a-9e76-9c21d0a01df4.png">

After I looked up some examples on Github, it looks like many projects use different folder structures or the default `tests` folder.
For example, Microsoft/Playwright uses this structure: 
<img width="524" alt="Screenshot 2022-07-27 at 11 22 33" src="https://user-images.githubusercontent.com/80746311/181213869-f675d946-abb4-41c6-ba52-e4c6e9daedf7.png">

As an alternative, since Playwright doesn't use `e2e` as a folder name, we may use the default folder name `tests` for the test files. If we want to change the name because it is not descriptive enough, I will change it to `end-to-end` because it is the most understandable folder name for our students.

With the `end-to-end`:

<img width="313" alt="Screenshot 2022-07-27 at 14 39 09" src="https://user-images.githubusercontent.com/80746311/181249147-c33c21ad-26b0-477c-a57d-85af7d504703.png">

For the `playwright.config` file, there are just some small changes: 

```diff
import { devices, PlaywrightTestConfig } from '@playwright/test';

// config file reference - https://playwright.dev/docs/test-configuration
const config: PlaywrightTestConfig = {
  webServer: {
    command: 'yarn start',
    port: 3000,
    reuseExistingServer: !process.env.CI,
  },
  testIgnore: '**/util/__tests__/**',
  retries: process.env.CI ? 2 : 0,
-  reporter: process.env.CI ? 'list' : 'html',
+  reporter: process.env.CI
+    ? 'list'
+    : [['html', { outputFolder: 'playwright/report/' }]],
-  outputDir: 'test-results/',
+  outputDir: 'playwright/test-results/',

  use: {
    trace: 'on-first-retry',
    screenshot: 'only-on-failure',
    video: 'retain-on-failure',
  },

  projects: [
    {
      name: 'chromium',
      use: {
        ...devices['Desktop Chrome'],
      },
    },
  ],
};

export default config;
```

We don't have to configure a `testDir`, because Playwright defaults to the directory of the configuration file, recursively:

> Defaults to the directory of the configuration file.

ref: https://playwright.dev/docs/api/class-testconfig#test-config-test-dir
 
For the `Jest` test, we can ignore the whole Playwright folder

`testPathIgnorePatterns: ['<rootDir>/playwright/'],`

For the Jest directory structure, we won't change anything because:

1. creating a `jest` folder containing all Jest test files is not as common, not as discoverable and it is further away from the code being tested. 
2. creating `.test.ts` files next to the files being tested is not as organized, and not as consistent (not every file has a test file)

**Comment by karlhorky** [comment]
Does Playwright use the name `e2e` anywhere already? Or maybe they use `end-to-end` or `End to End` (or even `integration`) instead?

**Comment by ProchaLu** [comment]
1. Playwright doesn't use `e2e` anywhere. I took the structure from Cypress. When I look into other Github projects, there are many different structures.
For example, Microsoft/Playwright uses this structure: 
<img width="524" alt="Screenshot 2022-07-27 at 11 22 33" src="https://user-images.githubusercontent.com/80746311/181213869-f675d946-abb4-41c6-ba52-e4c6e9daedf7.png">

After I saw some examples, it looks like many projects use different folder structures or the default tests folder.
Since Playwright doesn't use `e2e` as a folder name, maybe we can use the default folder name `tests` for the test files. If we want to change the name I would change it to `end-to-end` because it is the most understandable folder name for our students.

2. Almost every project has a different folder structure or uses the generic `tests` folder. Maybe students don't catch the `tests` folder, so to make it more noticeable, I would make a `playwright` folder. 

3. The default values for the `report` and `test-results` folder would be the root, but since I would create a folder for all Playwright-related folders and files, I would add them to the folder.

<img width="316" alt="Screenshot 2022-07-27 at 12 18 43" src="https://user-images.githubusercontent.com/80746311/181223838-e8622149-9d18-4633-b1c5-32ff1c7fc814.png">
